EIPS/eip-6963.md

```diff
@@ -0,0 +1,121 @@
+---
+eip: 6963
+title: Multi Injected Provider Interface (aka MIPI)
+description: Adding `window.evmproviders` to replace `window.ethereum`
+author: Pedro Gomes (@pedrouid), Kosala Hemachandra (@kvhnuke), Richard Moore (@ricmoo), Gregory Markou (@GregTheGreek)
+discussions-to: https://ethereum-magicians.org/t/eip-6963-multi-injected-provider-interface-aka-mipi
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-05-01
+requires: 1193
+replaces: 5749
+---
+
+## Abstract
+
+A Javascript interface injected by browser extensions installed by the user that enables user selection from multiple wallet providers using a window object labelled `window.evemproviders` instead of overwriting `window.ethereum` object.
+
+## Motivation
+
+Currently, wallet providers that offer browser extensions must inject their Ethereum providers (EIP-1193) into the same window object `window.ethereum` however this creates conflicts for users that may install more than one browser extension.
+
+Browser extensions are loaded in the web page and do not have a predictable pattern resulting in a race condition where the user is not in control to choose which wallet provider must be selected for exposing Ehtereum interface under the `window.ethereum` object.
+
+This results not only in a degraded user experience but also increases the barrier to entry for new browser extensions as users are forced to only install one browser extension at a time.
+
+Some browser extensions attempt to counteract this problem by delaying their injection to overwrite the same `window.ethereum` object which creates an unfair competition for wallet providers and lack of interoperability.
+
+In this proposal, we provide a solution that optimitizes for interoperability and enables fairer competition by lowering the barrier to entry for new wallet providers to improve the user experience on Ethereum.
+
+This is achieved by replacing the current window object `window.ethereum` with a new window object `window.evmproviders` that enables multiple browser extensions to be exposed for user selection in the web page.
```

Comment:
I’m not too sure if we should be “replacing” `window.ethereum` as this will break every dapp and I’d say we want to avoid telling consumers to update their dapp by X date… 😅

IMO, `window.ethereum` has served us quite well and I don’t believe we should be replacing it entirely to solve an issue that affects a niche audience (pro-users & developers). Most “mainstream” users wouldn’t be using multiple browser extension wallets.

Reply:
Perhaps replacing is the wrong terminology here...

I meant more like prioritizing `window.evmproviders` usage over `window.ethereum`

You will notice a few lines after I mention how an Ethereum library can overwrite `window.ethereum` based on the user selection of a wallet provider

PS - I would avoid changing existing interfaces like `window.ethereum` since that would impact backwards compatiblity